SMLHelper/Patchers/CraftDataPatcher_Subnautica.cs

```diff
@@ -1,97 +1,301 @@
+            harmony.Patch(AccessTools.Method(typeof(CraftData), nameof(CraftData.GetHarvestFinalCutBonus)),
+               prefix: new HarmonyMethod(AccessTools.Method(typeof(CraftDataPatcher), nameof(GetFinalCutBonusPrefix))));
+            harmony.Patch(AccessTools.Method(typeof(CraftData), nameof(CraftData.GetItemSize)),
+               prefix: new HarmonyMethod(AccessTools.Method(typeof(CraftDataPatcher), nameof(GetItemSizePrefix))));
+            harmony.Patch(AccessTools.Method(typeof(CraftData), nameof(CraftData.GetEquipmentType)),
+               prefix: new HarmonyMethod(AccessTools.Method(typeof(CraftDataPatcher), nameof(GetEquipmentTypePrefix))));
+            harmony.Patch(AccessTools.Method(typeof(CraftData), nameof(CraftData.GetQuickSlotType)),
+               prefix: new HarmonyMethod(AccessTools.Method(typeof(CraftDataPatcher), nameof(GetSlotTypePrefix))));
+            harmony.Patch(AccessTools.Method(typeof(CraftData), nameof(CraftData.GetCraftTime)),
+               prefix: new HarmonyMethod(AccessTools.Method(typeof(CraftDataPatcher), nameof(GetCraftTimePrefix))));
+            harmony.Patch(AccessTools.Method(typeof(CraftData), nameof(CraftData.GetCookedData)),
+               prefix: new HarmonyMethod(AccessTools.Method(typeof(CraftDataPatcher), nameof(GetCookedCreaturePrefix))));
+            harmony.Patch(AccessTools.Method(typeof(CraftData), nameof(CraftData.GetBackgroundType)),
+               prefix: new HarmonyMethod(AccessTools.Method(typeof(CraftDataPatcher), nameof(GetBackgroundTypesPrefix))));
+            harmony.Patch(AccessTools.Method(typeof(CraftData), nameof(CraftData.IsBuildableTech)),
+               prefix: new HarmonyMethod(AccessTools.Method(typeof(CraftDataPatcher), nameof(GetBuildablePrefix))));
+            
+            IngameMenuHandler.Main.RegisterOnQuitEvent(() => ResetAltered());
 
-            AddCustomTechDataToOriginalDictionary();
         }
 
-        private static void AddCustomTechDataToOriginalDictionary()
+        private static void ResetAltered()
+        {
+            TechDataAltered = 0;
+            HarvestOutputAltered = 0;
+            HarvestTypeAltered = 0;
+            FinalCutBonusAltered = 0;
+            ItemSizesAltered = 0;
+            EquipmentTypesAltered = 0;
+            SlotTypesAltered = 0;
+            CraftingTimesAltered = 0;
+            CookedCreatureAltered = 0;
+            BackgroundTypesAltered = 0;
+            BuildablesAltered = 0;
+            PrefabAltered = 0;
+        }
+
+        private static void CraftDataGetPrefabPrefix()
+        {
+            if (ModPrefab.PrefabCount > 0)
+            {
+                ModPrefab prefab = ModPrefab.Prefabs.FirstOrFallback(null);
+                if (!CraftData.techMapping.ContainsKey(prefab.TechType) || !CraftData.entClassTechTable.ContainsKey(prefab.ClassID))
+                    CraftDataPrefabIDCachePostfix();
```

Comment:
Can you add a comment explaining why the Prefix patch method is invoking the Postfix patch method?

Maybe also explain it here as well.

Reply:
Will fix the naming because the prefix is kind of a verify patch required and the postfix is do it anyway.